app/views/booking_requests/slots_step.html.erb

```diff
@@ -1,33 +1,33 @@
 <% content_for :header, t('.title') %>
+<%= render('shared/slotpicker_templates') %>
 
-<div class="Grid">
-  <div class="Grid-2-3">
     <%= form_for(@steps.fetch(:slots_step), url: booking_requests_path) do |f| %>
       <%= render(partial: 'hidden_prisoner_step') %>
       <%= render(partial: 'hidden_visitors_step') %>
 
-      <fieldset>
-        <% 3.times do |i| %>
-          <%=
-            single_field(
-              f, :"option_#{i}", :select,
-              f.object.available_slots.map { |s|
-                [format_slot_begin_time_for_public(s), s.iso8601]
-              },
-              include_blank: true
-            )
-          %>
-        <% end %>
+  <div class="SlotPicker" data-today="<%= Time.zone.today %>">
+    <div class="Grid visible--js-enabled">
+      <div class="Grid-1-2">
+        <fieldset>
+          <%= render('booking_calendar', prison: f.object.prison) %>
+          <%= render('date_slider') %>
+          <%= render('booking_calendar_legend') %>
+          <%= render('slot_picker_time_slots') %>
```

Comment:
At some point, we should rationalise our use of `render(…)` vs `render(:partial, …)`. I have a feeling that the `:partial` bit is probably just noise.

But that's not for this PR.